train-controller/src/main/java/hu/bme/mit/train/controller/TrainControllerImpl.java

```diff
@@ -41,15 +41,30 @@ private void enforceSpeedLimit() {
 		}
 	}
 
+	@Override
+    public void run() {
+        while (true) {
+            try {
+                // Call the followSpeed method
+                this.followSpeed();
+                Thread.sleep(10); // Sleep for 10 milliseconds
+            } catch (InterruptedException e) {
+                e.printStackTrace();
+            }
+        }
+    }
+
 	@Override
 	public void setJoystickPosition(int joystickPosition) {
 		this.step = joystickPosition;
-		
+		TrainControllerImpl thread = new TrainControllerImpl();
+		thread.start();
```

Comment:
you want to run this in the TrainControllerImpl constructor so it only runs once, rather than here